staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -252,106 +274,149 @@ func getNestedMap(obj map[string]interface{}, fields ...string) map[string]strin
 	return nil
 }
 
-func setNestedField(obj map[string]interface{}, value interface{}, fields ...string) {
+// NestedMap returns the map[string]interface{} value of a nested field.
+// Returns nil if value is not found or is not a map[string]interface{}.
+func NestedMap(obj map[string]interface{}, fields ...string) map[string]interface{} {
+	if m, ok := NestedField(obj, fields...).(map[string]interface{}); ok {
+		nMap := make(map[string]interface{}, len(m))
+		for k, v := range m {
+			nMap[k] = v
+		}
+		return nMap
+	}
+	return nil
+}
+
+// SetNestedField sets the value of a nested field.
+func SetNestedField(obj map[string]interface{}, value interface{}, fields ...string) {
 	m := obj
-	if len(fields) > 1 {
-		for _, field := range fields[0 : len(fields)-1] {
-			if _, ok := m[field].(map[string]interface{}); !ok {
-				m[field] = make(map[string]interface{})
-			}
-			m = m[field].(map[string]interface{})
+	for _, field := range fields[:len(fields)-1] {
+		if _, ok := m[field].(map[string]interface{}); !ok {
+			m[field] = make(map[string]interface{})
```

Comment:
this is unexpected that everything of wrong type is overridden.

Reply:
Also here we need errors.

Reply:
Well, I agree with you. This is how it is implemented right now, I just made it public. I'm happy to change it.
Do you want all of these functions to return an `error`? Maybe `bool` for getters?

Reply:
`bool` is also fine, maybe even better.